pkg/block/index.go

```diff
@@ -348,7 +367,12 @@ func GatherIndexIssueStats(logger log.Logger, fn string, minTime int64, maxTime
 		l0 := lset[0]
 		for _, l := range lset[1:] {
 			if l.Name < l0.Name {
-				return stats, errors.Errorf("out-of-order label set %s for series %d", lset, id)
+				stats.OutOfOrderLabels++
+				level.Warn(logger).Log("msg",
+					"out-of-order label set: known bug in Prometheus 2.8.0 and below",
```

Comment:
Perhaps we should suggest users here to enable `--debug.accept-malformed-index` here? I mean we would only tell users about a known bug now but wouldn't tell them how to fix it so that's weird 😄 we could be more user-friendly.

Reply:
I addressed this in `pkg/compact/compact.go` where the error is generated that would otherwise halt execution of the compact component.  Rather than here were it just becomes noise when the CLI option is enabled.

Reply:
Good point, that's even better :+1: